osu.Game/Online/Multiplayer/ServerShuttingDownCountdown.cs

```diff
@@ -0,0 +1,20 @@
+// Copyright (c) ppy Pty Ltd <contact@ppy.sh>. Licensed under the MIT Licence.
+// See the LICENCE file in the repository root for full licence text.
+
+using MessagePack;
+
+namespace osu.Game.Online.Multiplayer
+{
+    /// <summary>
+    /// A countdown that indicates the current multiplayer server is shutting down.
+    /// </summary>
+    [MessagePackObject]
+    public class ServerShuttingDownCountdown : MultiplayerCountdown
```

Comment:
Looking back at this, it's a touch unfortunate this is multiplayer specific (ie. doesn't cover the spectator server scenario). But I guess it's fine for now.

Reply:
Technically we should be able to send the countdown to all users connected to the server. Can't use the existing `SeverMultiplayerRoom` implementation for that but it can be transmitted directly over `MatchEvent()`.

Not sure it's necessary for now tho?